R/yaml-as-df.R

```diff
@@ -53,10 +65,23 @@ yaml_as_df <- function(path, quiet = FALSE) { #nocov start
   if(rename) {
     names(ans) <- prototype_names
   }
+
   if(exists("cols", meta)) {
     cols <- meta[["cols"]]
-    ans <- select(ans, cols, tidyselect::everything())
+    ans <- select(ans, all_of(cols), everything())
   }
+
+  if(is.character(row_var)) {
+    row_var0 <- row_var
+    i <- 1
+    while(row_var %in% names(ans)) {
```

Comment:
I like that this takes care to avoid a collision, but, as a user, I'd prefer an error here so that I don't get any surprises if my downstream code tries to use `row_name` while it really should be using, say, `row_name_1` due to a collision.

If the default of `row_var` was `NULL`, then that could be done without worrying about breaking existing code.  But given the default of `row_var = ".row"`, it'd be too risky.

Considering that ".row" is a nicer default (so that this behavior isn't disabled by default) and that collisions are likely to be rare, this logic to avoid the collision is probably the way to go.

Reply:
I wondered if this would be overkill.  

How about a message when we re-name the column with row names? I think it should be rare to see the message; but it would let the function to succeed and alert the user about it.

Reply:
Yeah, a message or warning seems reasonable given it's expected to be rare.